src/encoded/static/components/image.js

```diff
@@ -5,177 +5,149 @@ import * as globals from './globals';
+    const lightbox = React.useRef(null);
+
+    // Called when the window is resized to resize the image within it.
+    const handleResize = () => {
+        setImgHeight(lightbox.current.offsetHeight - 40);
+    };
+
+    // Called when a key is pressed to close the lightbox with ESC.
+    const handleKeyDown = (e) => {
+        if (e.keyCode === 27) {
+            clearLightbox();
+        }
+    };
+
+    React.useEffect(() => {
+        // Need to resize and attach event handlers at mount.
+        handleResize();
+        window.addEventListener('resize', handleResize);
+        return (() => {
+            window.removeEventListener('resize', handleResize);
+        });
+    }, []);
+
+    const lightboxClass = `lightbox${lightboxVisible ? ' active' : ''}`;
+    return (
+        <div tabIndex="-1" className={lightboxClass} onClick={clearLightbox} onKeyDown={handleKeyDown} role="button" aria-label="Close" ref={lightbox}>
+            <div className="lightbox-img">
+                <a aria-label="Open image" data-bypass="true" href={lightboxImg}>
+                    <img src={lightboxImg} alt="Attachment from submitters" style={{ maxHeight: imgHeight }} />
+                </a>
+                <button className="lightbox-close" aria-label="Close" onClick={clearLightbox} />
             </div>
-        );
-        /* eslint-enable jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions */
-    }
-}
+        </div>
+    );
+};
 
 Lightbox.propTypes = {
-    lightboxVisible: PropTypes.bool, // True if lightbox should be rendered or hidden initially
-    clearLightbox: PropTypes.func.isRequired, // Callback function to hide light box
-    lightboxImg: PropTypes.string.isRequired, // URI of image to display in the light box
+    /** True if lightbox should be rendered or hidden initially */
+    lightboxVisible: PropTypes.bool,
+    /** Callback function to hide light box */
+    clearLightbox: PropTypes.func.isRequired,
+    /** URI of image to display in the light box */
+    lightboxImg: PropTypes.string.isRequired,
 };
 
 Lightbox.defaultProps = {
     lightboxVisible: false,
 };
 
 
-export class Attachment extends React.Component {
-    constructor() {
-        super();
-
-        // Set initial React state.
-        this.state = { lightboxVisible: false };
-
-        // Bind this to non-React methods.
-        this.lightboxClickImage = this.lightboxClickImage.bind(this);
-        this.clearLightbox = this.clearLightbox.bind(this);
-        this.handleEscKey = this.handleEscKey.bind(this);
-    }
-
-    // Register for keyup events for ESC key
-    componentDidMount() {
-        globals.bindEvent(window, 'keyup', this.handleEscKey);
-    }
-
-    // Unregister keyup events when component closes
-    componentWillUnmount() {
-        globals.unbindEvent(window, 'keyup', this.handleEscKey);
-    }
+export const Attachment = ({ context, attachment, className, showLink }) => {
```

Comment:
"showLink"- show what link?
"className"- what affect does the css class have?

Reply:
showLink: “False to just display image preview without link or lightbox” from the comment in propTypes.

className: “CSS class name to add to image element; '-img' added to it”.

Reply:
It was rhetorical. :)